### PR TITLE
[JENKINS-42846] - Fix regressions in images after slave.jar => agent.jar renaming with improper symlink

### PR DESCRIPTION
**Pull request by oleg-nenashev**
Fixes agents. 3.35-1 is completely broken, and emergency reviews will be appreciated

### CODE REVIEW

File: Dockerfile-alpine [hunk 1]
```diff
@@ -36,9 +36,10 @@ LABEL Description="This is a base image, which provides the Jenkins agent execut
 ARG AGENT_WORKDIR=/home/${user}/agent
 
 RUN apk add --update --no-cache curl bash git git-lfs openssh-client openssl procps \
-  && curl --create-dirs -fsSLo /usr/share/jenkins/slave.jar https://repo.jenkins-ci.org/public/org/jenkins-ci/main/remoting/${VERSION}/remoting-${VERSION}.jar \
+  && curl --create-dirs -fsSLo /usr/share/jenkins/agent.jar https://repo.jenkins-ci.org/public/org/jenkins-ci/main/remoting/${VERSION}/remoting-${VERSION}.jar \
   && chmod 755 /usr/share/jenkins \
-  && chmod 644 /usr/share/jenkins/slave.jar \
+  && chmod 644 /usr/share/jenkins/agent.jar \
+  && ln -sf /usr/share/jenkins/agent.jar /usr/share/jenkins/slave.jar \
```
Comment (oleg-nenashev) [inline Dockerfile-alpine]: `-f` flag here was a bad idea, yes

Comment (jimklimov) [reply Dockerfile-alpine]: I think you don't really want full pathnames here, if you reference an `agent.jar` located in same directory as the `slave.jar` symlink. Full names make work with alternate root FSes (backups, look into containers from hosts, etc.) problematic, relative symlinks ease this pain a lot.

Comment (oleg-nenashev) [reply Dockerfile-alpine]: I suggest doing it as a follow-up. Just fixing the images now so that they do not fall apart

Comment (jimklimov) [reply Dockerfile-alpine]: Yes, sure, this nit is not urgent :)

Comment (stopalopa) [reply Dockerfile-alpine]: I now can't remember why I added the -f flag, but maybe it does something different than what I originally thought. Why is it a bad idea?

**Comment by oleg-nenashev** [comment]
-f flag "remove existing destination files", and hence it suppressed the
issue in this case (wrong argument ordering)

**Comment by stopalopa** [comment]
ln works differently in Mac and Linux?  When I check the man pages on my Mac it says the following for -f: If the target file already exists, then unlink it so that the link may occur.  (The -f option overrides any previous -i options.)

**Pull request merged**